palette/Cargo.toml

```diff
@@ -39,14 +43,20 @@ features = ["serde_derive"]
 optional = true
 
 [dev-dependencies]
-image = "0.22"
-clap = "2"
 csv = "1"
+lazy_static = "1"
 serde = "1"
 serde_derive = "1"
-lazy_static = "1"
 serde_json = "1"
 
+[dev-dependencies.clap]
+version = "2"
+default-features = false
+
+[dev-dependencies.image]
+version = "0.22"
+default-features = false
+
```

Comment:
Good call! It's easy to forget that there may be features to disable. This may be something to look into for the rest of the dependencies, as I haven't done that in quite a while.

Reply:
### [palette](https://github.com/Ogeon/palette/blob/master/palette/Cargo.toml#L16) cargo.toml
[image](https://github.com/image-rs/image/blob/master/Cargo.toml) was definitely the heaviest: 
```toml
[features]
default = ["gif_codec", "jpeg", "ico", "png_codec", "pnm", "tga", "tiff", "webp", "bmp", "hdr", "dxt", "jpeg_rayon"]
```
[clap](https://github.com/clap-rs/clap/blob/v2-master/Cargo.toml) not as much since I realize master is on the v3 beta, this is v2.33
```toml
[features]
default = ["suggestions", "color", "vec_map"]
```
[serde](https://github.com/serde-rs/serde/blob/master/serde/Cargo.toml) and [serde_json](https://github.com/serde-rs/json/blob/master/Cargo.toml) not much to change since this is also feature gated on `palette`.
```toml
[features]
default = ["std"]
```
[phf](https://github.com/sfackler/rust-phf/blob/master/phf/Cargo.toml), similar to above
```toml
[features]
default = ["std"]
std = ["phf_shared/std"]
```
No default:
[lazy_static](https://github.com/rust-lang-nursery/lazy-static.rs/blob/master/Cargo.toml) 
[csv](https://github.com/BurntSushi/rust-csv/blob/master/Cargo.toml)
[serde_derive](https://github.com/serde-rs/serde/blob/master/serde_derive/Cargo.toml)
[phf_codegen](https://github.com/sfackler/rust-phf/blob/master/phf_codegen/Cargo.toml)

### [palette_derive](https://github.com/Ogeon/palette/blob/master/palette_derive/Cargo.toml#L17) cargo.toml
```toml
[dependencies]
syn = { version = "^1.0", features = ["extra-traits"] }
quote = "^1.0"
proc-macro2 = "^1.0"
```

[syn](https://github.com/dtolnay/syn/blob/master/Cargo.toml)
```toml
[features]
default = ["derive", "parsing", "printing", "clone-impls", "proc-macro"]
```
[quote](https://github.com/dtolnay/quote/blob/master/Cargo.toml)
```toml
[features]
default = ["proc-macro"]
# Disabling the proc-macro feature removes the dynamic library dependency on
# libproc_macro in the rustc compiler.
proc-macro = ["proc-macro2/proc-macro"]
```
[proc-macro2](https://github.com/alexcrichton/proc-macro2/blob/master/Cargo.toml)
```toml
[features]
proc-macro = []
default = ["proc-macro"]
```

----
I can remove `syn`'s `"printing",  "clone-impls"` features which don't reduce crate count but might improve compile time.

Reply:
Nice, seems like it's in a good shape now, after your change. The syn changes have to be measured to know the impact, but I guess it doesn't hurt to remove them if they are not used.

Reply:
Looks like I'll have to revert the syn changes, when I tried building locally I don't think it built cleanly.

Reply:
Ah, oh well. I think it's good anyway. I know there are other tricks, like not optimizing particular dependencies, but I don't think we need to go down that rabbit hole in this PR.